src/Symfony/Component/Console/Application.php

```diff
@@ -975,7 +977,18 @@ protected function doRunCommand(Command $command, InputInterface $input, OutputI
             }
         }
 
-        $commandSignals = $command instanceof SignalableCommandInterface ? $command->getSubscribedSignals() : [];
+        // bind before getSubscribedSignals() and the console.command event, so the methods and listeners have access to input options/arguments
+        try {
+            $command->mergeApplicationDefinition();
+            $input->bind($command->getDefinition());
+        } catch (ExceptionInterface) {
+            // ignore invalid options/arguments for now, to allow the event listeners to customize the InputDefinition
+        }
+
+        $commandSignals = $command instanceof SignalableCommandInterface ? $command->getSubscribedSignals($input, $output) : [];
+        if ($command instanceof AlarmableCommandInterface && \defined('SIGALRM') && SignalRegistry::isSupported() && !\in_array(\SIGALRM, $commandSignals, true)) {
```

Comment:
Making the TraceableCommand implement AlarmableCommandInterface means that in dev, **all** commands will register this `SIGALRM` (even when they are not alarmable). Is this expected ?

Reply:
No, good catch, I've totally missed that one. I've added a `TraceableAlarmableCommand`, so this should no longer be an issue. I'm not sure if this is the best approach, but I couldn't think of any other.